### DISCUSSION TITLE
Where are the props defined on a model actually processed and applied to an instance on .create()?

### DISCUSSION THREAD

**Discussion by techknowfile** [mobxjs/mobx-state-tree#1734] [Q&A] [answered]
Trying to replicate the logic for defining new Models by extending existing types with .actions, .views, and .props for a MobX store, where it won't use mobx-state-tree's typing (nor typescript). I understand how .actions() and .views() work, but am struggling to find the relevant code for where the [properties assigned to a Type](https://github.com/mobxjs/mobx-state-tree/blob/ac5ffcd84303a1664a4f3e073c384de8835d566f/packages/mobx-state-tree/src/types/complex-types/model.ts#L349) get applied to an actual instance of an object

**Comment by jamonholmgren** [comment, marked as answer]
This is where the actual application happens:

https://github.com/mobxjs/mobx-state-tree/blob/ac5ffcd84303a1664a4f3e073c384de8835d566f/packages/mobx-state-tree/src/core/node/object-node.ts#L133

But this code is a bit difficult to follow, to be honest. I wish it was less complicated.
